### ISSUE TITLE
False positive on `if pos { 0.0 } else { -0.0 }`

### ISSUE THREAD

**Issue by dtolnay** [Manishearth/rust-clippy#1086]
```
serde-rs/json/json/src/de.rs:438:47: 438:55 error: this `if` has identical blocks
serde-rs/json/json/src/de.rs:438         visitor.visit_f64(if pos { 0.0 } else { -0.0 })
                                                                               ^~~~~~~~
```

These are not identical.

**Comment by dtolnay** [comment]
For the benefit of search engines I should mention that one of them is "negative zero."

```rust
fn main() {
    is_positive(if true { 0.0 } else { -0.0 });  // true
    is_positive(if false { 0.0 } else { -0.0 }); // false
}

fn is_positive(f: f64) {
    println!("{}", f.is_sign_positive());
}
```

**Comment by mcarton** [comment]
Hum, we compare `f64` directly (and yes, `0.0 == -0.0`). Maybe we should transmute to `u64` and compare that.
